### ISSUE TITLE
ChatGLM2 6B 推理时候利用率不高，但是CPU占满了

### ISSUE THREAD

**Issue by yuanphoenix** [ModelTC/lightllm#134]
chatglm2 在推理时候，gpu 利用率不是很高，只有六七十。但是cpu占用达到了 100% 。起初怀疑是 tokenizer 的问题，但是发现decode和encode 是在两个进程中完成的，那两个进程cpu利用率几乎可以忽略。后来发现推理进程在  logits = self.model.forward(**kwargs)  过程中会占满 cpu ，替换这个语句，cpu占用只有十几。我觉得可能是CPU限制了速度？

forward的过程不应该在GPU上吗，为什么还会占用这么多cpu呢？

我的显卡是 3090 24G，ubuntu系统。CPU 是     Intel(R) Xeon(R) Gold 6226R CPU @ 2.90GHz

**Comment by yuanphoenix** [comment]
https://github.com/openai/triton/issues/416
是否和这个问题有关？

**Comment by hiworldwzj** [comment]
@yuanphoenix 应该和你说的那个问题没关系，我猜测是不是max_total_token_num 设置得比较小，同时打的并发比较小，导致gpu使用率不够高的？

**Comment by hiworldwzj** [comment]
@yuanphoenix  chatglm2采用的MQA 架构，可以支持很大的max_total_token_num.
